### PR TITLE
Add API to report disconnected endpoint count to ServiceControl.Monitoring

### PR DESCRIPTION
**Pull request by boblangley**
Adds a new API and test to report the total endpoint disconnected count for use by ServicePulse.

### CODE REVIEW

**Comment by WilliamBZA** [comment]
Don't forget milestones and labels

**Comment by WilliamBZA** [comment]
I've added to the 4.12.0 release.

File: src/ServiceControl.Monitoring.AcceptanceTests/Tests/When_querying_disconnected_count.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                }))
+                .WithEndpoint<MonitoredEndpoint>(b =>
+                b.CustomConfig(endpointConfig => endpointConfig.EnableMetrics().SendMetricDataToServiceControl(Settings.DEFAULT_ENDPOINT_NAME, TimeSpan.FromMilliseconds(200), "Second"))
+                .ToCreateInstance(endpointConfig => Endpoint.Create(endpointConfig), async startableEndpoint =>
+                {
+                    context.SecondInstance = await startableEndpoint.Start();
+
+                    return context.SecondInstance;
+                }))
+                .Done(async c =>
```
Comment (boblangley) [inline src/ServiceControl.Monitoring.AcceptanceTests/Tests/When_querying_disconnected_count.cs:39]: Remember this is run over and over until it returns true.

File: src/ServiceControl.Monitoring.AcceptanceTests/Tests/When_querying_disconnected_count.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    }
+
+                    c.AfterSecondStoppedCount = disconnectedCount;
+                    return true;
+                })
+                .Run();
+
+            Assert.AreEqual(0, context.AfterAllStartedCount, "Disconnected count after all endpoints started");
+            Assert.AreEqual(1, context.AfterFirstStoppedCount, "Disconnected count after first endpoint stopped");
+            Assert.AreEqual(2, context.AfterSecondStoppedCount, "Disconnected count after both endpoints stopped");
```
Comment (boblangley) [inline src/ServiceControl.Monitoring.AcceptanceTests/Tests/When_querying_disconnected_count.cs:84]: Are asserts are incorrect, but I think the rest of the test is correct.

**Pull request merged**